### PR TITLE
railway-travel: init at 2.4.0

### LINKED ISSUE #299285: Package request: railway
**Project description**

Railway lets you look up travel information for many different railways, all without needing to navigate through different websites.

**Metadata**

* homepage URL: https://mobile.schmidhuberj.de/railway
* source URL: https://gitlab.com/schmiddi-on-mobile/railway
* license: gpl3
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Lilacious**
source URL: https://gitlab.com/schmiddi-on-mobile/railway

This closes  #299285.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by wegank — DISMISSED** [review]
Please choose another attribute name: https://search.nixos.org/packages?channel=23.11&show=railway&from=0&size=50&sort=relevance&type=packages&query=railway

**Comment by Lilacious** [comment]
@wegank Is something like railway-travel ok?

**Review by Lilacious — COMMENTED** [review]
Resolved

**Comment by Lilacious** [comment]
Makes sense! thank you.

**Comment by wegank** [comment]
@ofborg build railway-travel railway-travel.passthru.tests

File: pkgs/by-name/ra/railway-travel/package.nix [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.Security
+    ];
+
+  meta = with lib; {
+    description = "Find all your travel information";
+    homepage = "https://gitlab.com/schmiddi-on-mobile/railway";
+    changelog = "https://gitlab.com/schmiddi-on-mobile/railway/-/blob/${src.rev}/CHANGELOG.md";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [lilacious];
```
Comment (wegank) [inline pkgs/by-name/ra/railway-travel/package.nix:67]:
```suggestion
    maintainers = with maintainers; [ lilacious ];
```

See https://github.com/NixOS/rfcs/blob/master/rfcs/0166-nix-formatting.md#attribute-sets-and-lists. Are you using Alejandra?

Comment (Lilacious) [reply pkgs/by-name/ra/railway-travel/package.nix:67]: Yes I am 😅

File: pkgs/by-name/ra/railway-travel/package.nix [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    [
+      cairo
+      gdk-pixbuf
+      glib
+      gtk4
+      libadwaita
+      pango
+    ]
+    ++ lib.optionals stdenv.isDarwin [
+      darwin.apple_sdk.frameworks.Security
```
Comment (wegank) [inline pkgs/by-name/ra/railway-travel/package.nix:59]: Probably not enough here. Some `env.NIX_CFLAGS_COMPILE` might also be needed on darwin, I'll look into it tonight.

Comment (Lilacious) [reply pkgs/by-name/ra/railway-travel/package.nix:60]: I just tried to build and run it on a x86_64 ventura vm and it worked without any issues.

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 299999` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>railway-travel</li>
  </ul>
</details>

**Review by wegank — APPROVED** [review]

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 299999` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>railway-travel</li>
  </ul>
</details>

**Review by dotlambda — COMMENTED** [review]
Why is the formatting in this file so inconsistent?

**Comment by Lilacious** [comment]
I'm not quite sure what you mean by that

**Comment by wegank** [comment]
I'm not a big fan of the list formatting in the file either, but it turns out to be stable under `nixfmt-rfc-style`...

File: pkgs/by-name/ra/railway-travel/package.nix [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  };
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-yalFC7Pw9rq1ylLwoxLi4joTyjQsZJ/ZC61GhTNc49w=";
+  };
+
+  nativeBuildInputs = [
+    desktop-file-utils
```
Comment (dotlambda) [inline pkgs/by-name/ra/railway-travel/package.nix:39]: This ...

File: pkgs/by-name/ra/railway-travel/package.nix [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    ninja
+    pkg-config
+    rustPlatform.cargoSetupHook
+    rustc
+    wrapGAppsHook4
+  ];
+
+  buildInputs =
+    [
+      cairo
```
Comment (dotlambda) [inline pkgs/by-name/ra/railway-travel/package.nix:51]: ... vs this

File: pkgs/by-name/ra/railway-travel/package.nix [hunk 5]
```diff
@@ -0,0 +1,76 @@
+{
+  lib,
```
Comment (dotlambda) [inline pkgs/by-name/ra/railway-travel/package.nix:2]: I also don't like this style but that's besides the point.

Comment (Lilacious) [reply pkgs/by-name/ra/railway-travel/package.nix:2]: Thanks for pointing it out. I didn't even notice that if I'm honest, I will pay more attention to it next time.

Comment (wegank) [reply pkgs/by-name/ra/railway-travel/package.nix:2]: This is the style in RFC 166, so not your fault.

**Pull request merged**